### PR TITLE
[AUD-1812] Prevent unnecessary store syncs

### PR DESCRIPTION
**Pull request by sliptype**
### Description

* Updates some reducers to not update state when not necessary. This allows the equality checks to be more accurate when determining when to sync state to the mobile layer
* Removes a todo and special case for `listen-streak` challenge types. See: https://linear.app/audius/issue/AUD-1536/fix-challenge-overrides-for-completed-challenges

### Dragons

Touches rewards reducers which could be dangerous

### How Has This Been Tested?

iOS sim, checked that empty sync actions were reduced

### How will this change be monitored?

TestFlight QA

### CODE REVIEW

File: packages/web/src/common/store/pages/audio-rewards/slice.ts [hunk 1]
```diff
@@ -97,18 +97,13 @@ const slice = createSlice({
...
-          }
-          acc[challenge.challenge_id] = challenge
-          return acc
-        }, {} as Partial<Record<ChallengeRewardID, UserChallenge>>)
+        state.userChallenges = userChallenges.reduce(
+          (acc, challenge) => ({
+            ...acc,
+            [challenge.challenge_id]: challenge
+          }),
+          {} as Partial<Record<ChallengeRewardID, UserChallenge>>
```
Comment (dylanjeffers) [inline packages/web/src/common/store/pages/audio-rewards/slice.ts:105]: not something you introduced, but can we type the `acc` or the reduce<> with this object type so we can prevent a cast?

**Review by dylanjeffers — APPROVED** [review]
awesome job figuring this out

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/sk-aud-1812-prevent-empty-sync-actions

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/sk-aud-1812-prevent-empty-sync-actions

**Review by piazzatron — APPROVED** [review]

**Review by rickyrombo — APPROVED** [review]

**Pull request merged**